webapp/src/Controller/API/JudgehostController.php

```diff
@@ -185,7 +185,9 @@ public function createJudgehostAction(Request $request): array
             ->andWhere('j.judgingid = jt.jobid')
             ->andWhere('jr.runresult IS NULL')
             ->andWhere('j.valid = 1 OR r.valid = 1')
+            ->andWhere('j.result != :compiler_error')
```

Comment:
Shouldn't this just check for `result NOT IS NULL`?

Reply:
I assume you mean `j.result IS NULL` (so no `NOT`).

This doesn't work when we judge remaining testcases of a problem after the final verdict is known and crash while doing so.

Reply:
Indeed, I meant `j.result IS NULL`. To be honest, I think that we have to think better about how to do this kind of logic: as a query or code, do we want to move certain logic away in common functions, completely rethinnk this, etc. as I feel that these are places that contain critical code, but are getting too complex.